cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -440,26 +441,15 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 	// finally, filter out pods that are too "young" to safely be considered for a scale-up (delay is configurable)
 	unschedulablePodsToHelp = a.filterOutYoungPods(unschedulablePodsToHelp, currentTime)
 
-	if len(unschedulablePodsToHelp) == 0 {
-		scaleUpStatus.Result = status.ScaleUpNotNeeded
-		klog.V(1).Info("No unschedulable pods")
-	} else if a.MaxNodesTotal > 0 && len(readyNodes) >= a.MaxNodesTotal {
-		scaleUpStatus.Result = status.ScaleUpNoOptionsAvailable
-		klog.V(1).Info("Max total nodes in cluster reached")
-	} else if allPodsAreNew(unschedulablePodsToHelp, currentTime) {
-		// The assumption here is that these pods have been created very recently and probably there
-		// is more pods to come. In theory we could check the newest pod time but then if pod were created
-		// slowly but at the pace of 1 every 2 seconds then no scale up would be triggered for long time.
-		// We also want to skip a real scale down (just like if the pods were handled).
-		a.processorCallbacks.DisableScaleDownForLoop()
-		scaleUpStatus.Result = status.ScaleUpInCooldown
-		klog.V(1).Info("Unschedulable pods are very new, waiting one iteration for more")
-	} else {
+	scaleUpResourceManager := scaleup.NewResourceManager(a.processors.CustomResourcesProcessor)
```

Comment:
I think now this can be created once, not just once per loop. Perhaps it would make sense to move it to a field in `StaticAutoscaler`?

Reply:
Yeah, this makes sense, as the only dependency `CustomResourcesProcessor` is also created and initialized there.